src/pages/iv-metals/forms/ScrapForm.js

```diff
@@ -0,0 +1,178 @@
+import { useContext, useEffect } from 'react'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { ControlContext } from 'src/providers/ControlContext'
+
+const ScrapForm = ({ store, setStore, maxAccess, labels }) => {
+  const { recordId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    enableReinitialize: true,
+    validateOnChange: true,
+    maxAccess,
+    validationSchema: yup.object({
+      scrap: yup
+        .array()
+        .of(
+          yup.object().shape({
+            sku: yup.string().required(),
+            purity: yup
+              .number()
+              .nullable()
+              .test('is-valid-reportingPurity', function (value) {
+                return value >= 0.001 && value <= 1 ? true : false
+              })
+          })
+        )
+        .required()
+    }),
+
+    initialValues: {
+      recordId: recordId,
+      scrap: [
+        {
+          id: 1,
+          metalId: recordId,
+          seqNo: '',
+          itemId: '',
+          sku: '',
+          itemName: '',
+          laborValuePerGram: '',
+          purity: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      postScrap(values)
+    }
+  })
+
+  const postScrap = obj => {
+    const items = obj?.scrap.map(({ id, ...item }) => ({
+      ...item,
+      metalId: recordId,
+      seqNo: id
+    }))
+
+    const data = {
+      metalId: recordId,
+      items: items
+    }
+
+    postRequest({
+      extension: InventoryRepository.Scrap.set2,
+      record: JSON.stringify(data)
+    })
+      .then(res => {
+        toast.success(platformLabels.Added)
+        setStore(prevStore => ({
```

Comment:
Could you please clarify why we're updating items in the store when it only contains recordId? If it's not necessary, please remove it, as well as from the props and any references in the window.